src/functionalTest/kotlin/com/vaadin/gradle/VaadinSmokeTest.kt

```diff
@@ -53,68 +56,99 @@ class VaadinSmokeTest : AbstractGradleTest() {
         // let's explicitly check that vaadinPrepareFrontend has been run.
         result.expectTaskSucceded("vaadinPrepareFrontend")
 
-        val build = File(testProjectDir, "build/resources/main/META-INF/VAADIN/build")
+        val build = File(testProject.dir, "build/resources/main/META-INF/VAADIN/webapp/VAADIN/build")
         expect(true, build.toString()) { build.exists() }
         build.find("*.gz", 5..10)
         build.find("*.js", 5..10)
 
-        val tokenFile = File(build, "../config/flow-build-info.json")
+        val tokenFile = File(testProject.dir, "build/resources/main/META-INF/VAADIN/config/flow-build-info.json")
         val buildInfo: JsonObject = JsonUtil.parse(tokenFile.readText())
-        expect(false, buildInfo.toJson()) { buildInfo.getBoolean(Constants.SERVLET_PARAMETER_ENABLE_DEV_SERVER) }
+        expect(false, buildInfo.toJson()) { buildInfo.getBoolean(Constants.SERVLET_PARAMETER_PRODUCTION_MODE) }
     }
 
     @Test
     fun testBuildFrontendInProductionMode() {
-        val result: BuildResult = build("-Pvaadin.productionMode", "vaadinBuildFrontend")
+        val result: BuildResult = testProject.build("-Pvaadin.productionMode", "vaadinBuildFrontend")
         // vaadinBuildFrontend depends on vaadinPrepareFrontend
         // let's explicitly check that vaadinPrepareFrontend has been run
         result.expectTaskSucceded("vaadinPrepareFrontend")
 
-        val build = File(testProjectDir, "build/resources/main/META-INF/VAADIN/build")
+        val build = File(testProject.dir, "build/resources/main/META-INF/VAADIN/webapp/VAADIN/build")
         expect(true, build.toString()) { build.isDirectory }
         expect(true) { build.listFiles()!!.isNotEmpty() }
         build.find("*.gz", 5..10)
         build.find("*.js", 5..10)
+        val tokenFile = File(testProject.dir, "build/resources/main/META-INF/VAADIN/config/flow-build-info.json")
+        val buildInfo: JsonObject = JsonUtil.parse(tokenFile.readText())
+        expect(true, buildInfo.toJson()) { buildInfo.getBoolean(Constants.SERVLET_PARAMETER_PRODUCTION_MODE) }
+    }
+
+    @Test
+    fun testBuildWarBuildsFrontendInProductionMode() {
+        testProject.newFile("src/main/java/org/vaadin/example/MainView.java", """
+            package org.vaadin.example;
+
+            import com.vaadin.flow.component.html.Span;
+            import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+            import com.vaadin.flow.router.Route;
+
+            @Route("")
+            public class MainView extends VerticalLayout {
+
+                public MainView() {
+                    add(new Span("It works!"));
+                }
+            }
+        """.trimIndent())
+
+        val result: BuildResult = testProject.build("-Pvaadin.productionMode", "build")
+        result.expectTaskSucceded("vaadinPrepareFrontend")
+        result.expectTaskSucceded("vaadinBuildFrontend")
+        val war = testProject.builtWar
```

Comment:
either we can comment 
```suggestion
        // val war = testProject.builtWar
```
or remove 
```suggestion
```
this line.

Reply:
Actually I would suggest to assert that the war file exists. I'll push appropriate code.

Reply:
(Resolved in master)